kong/core/certificate.lua

```diff
@@ -81,10 +81,16 @@ function _M.execute()
   -- retrieve SNI or raw server IP
 
   local sni, err = ssl.server_name()
-  if not sni then
-    log(ERR, "could not retrieve Server Name Indication from client: ", err)
+  if err then
+    log(ERR, "could not retrieve Server Name Indication: ", err)
     return ngx.exit(ngx.ERROR)
   end
+  if not sni then
+    log(DEBUG, "no Server Name Indication provided by client, serving ".. 
```

Comment:
Use a comma and the variadic form of this function to avoid string concatenation on the Lua land.